src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -173,8 +173,15 @@ private SearchHit[] getSearchHits(final List<Integer> docIds, final FetchSearchR
         SearchHits searchHits = fetchSearchResult.hits();
         SearchHit[] searchHitArray = searchHits.getHits();
         // validate the both collections are of the same size
-        if (Objects.isNull(searchHitArray) || searchHitArray.length != docIds.size()) {
-            throw new IllegalStateException("Score normalization processor cannot produce final query result");
+        if (Objects.isNull(searchHitArray)) {
+            throw new IllegalStateException(
```

Comment:
do we need this? when we are doing the same thing in this function shouldSkipProcessorDueToIncompatibleQueryAndFetchResults?

Reply:
yes, it's the same check. One thing is different - earlier in the processor code we're skipping execution, here we're throwing exception. Let's finalize on the approach mentioned in the earlier comment, do we skip processor or throw error. If we throw error I'll move these exceptions to processor

Reply:
removed the check from workflow class, let me know if we need to throw exceptions in processor class